src/core/cuda_device.h

```diff
@@ -0,0 +1,31 @@
+#ifndef CUDA_DEVICE_H
+#define CUDA_DEVICE_H
+#include "cuda_common.h"
+
+
+
+namespace CUDA
+{
```

Comment:
The description of an entire class needs to be a lot more than a single sentence. Reference my description an the OpenCL::Device class.